.travis.yml

```diff
@@ -4,26 +4,31 @@ os:
   - osx
 julia:
   - 0.6
+  - nightly
+matrix:
+  allow_failures:
+    - julia: nightly
 notifications:
   email: false
 sudo: false
 cache:
   directories:
-  - $HOME/usr
+    - /home/travis/.julia
 addons:
   apt_packages:
-  - gfortran
-env:
-  - MAKEFLAGS="-j3"
-    LD_LIBRARY_PATH=$HOME/usr/lib
-# Override the default test script
-# We need to add `Cbc`, `Ipopt` and `CoinOptServices` manually so we can cache
-# them after the tests have run
+    - gfortran
+
+before_install:
+  - julia -e 'if "AmplNLWriter" in keys(Pkg.installed()) Pkg.rm("AmplNLWriter"); Pkg.rm("AmplNLWriter") end'
+  - julia -e 'Pkg.update()' #make sure we get the latest version of METADATA
+  - julia -e 'if !("Ipopt" in keys(Pkg.installed())) Pkg.add("Ipopt") end'
+  - julia -e 'if !("Coverage" in keys(Pkg.installed())) Pkg.add("Coverage") end'
+
 script:
- - if [[ -a .git/shallow ]]; then git fetch --unshallow; fi
- - julia --check-bounds=yes -e 'Pkg.clone(pwd()); Pkg.build("AmplNLWriter"); Pkg.add("Ipopt"); Pkg.test("AmplNLWriter")'
-before_cache:
-  - cp -R $HOME/.julia/*/Ipopt/deps/usr $HOME
+  - if [[ -a .git/shallow ]]; then git fetch --unshallow; fi
+  - julia --check-bounds=yes -e 'Pkg.clone(pwd()); Pkg.build("AmplNLWriter"); Pkg.test("AmplNLWriter")'
```

Comment:
Can you change this to `Pkg.test("AmplNLWriter", coverage=true)` while you're at it? I forgot to do it when I added the .codecov.yml